WIP: RFC-10 Cache (#120)
tango-controls/rfc: 10/RequestReply.md

#### Cache

> The Cache is a circular buffer of recent responses for requests genereted internally by Device Server's Polling mechanism.

Reviewer: ```suggestion
The Cache is a circular buffer of recent responses for requests generated internally by Device Server's Polling mechanism.
```

The Polling is periodical invoking a list of requests on Devices instantiated by a Device Server. A period of invokiong a request is called a Polling Period. 

* The Cache SHALL be implemented by a Device Server.

* The Polling SHALL support the following requests:
  * read Attribute,

>   * execution of  Command with DevVoid argin,

Reviewer: ```suggestion
  * execution of  Command with DevVoid argin
```

> * The Polling Period MAY be configured separately for each Attribute or Command of each Device spearately.

Reviewer: ```suggestion
* The Polling Period MAY be configured separately for each Attribute or Command of each Device.
```

* A Device Class MAY implement default settings of the Polling for Attributes and Commands.

* A Client MAY change the Polling Period of any Attribute or Command of any Device.

* A Client MAY switch off/on a Polling for any Attribute or Command of any Device.

* The Polling SHALL be configurable by a Client via the following Admin Device commands:
  * `AddObjPolling`:

>     It add a new object (command or attribute) to the list of object(s) to be polled. It is also with this command that the polling period is specified.

Reviewer: ```suggestion
    Add a new object (command or attribute) to the list of object(s) to be polled. It is also with this command that the polling period is specified.
```

  * `RemObjPolling`:
    To remove one object (command or attribute) from the polled object(s) list
  * `UpdObjPollingPeriod`:
    Change one object polling period
  * `StartPolling`:
    Starts polling for the whole process
  * `StopPolling`:
    Stops polling for the whole process
  * `PolledDevice`:

>     Allow a client to know which device are polled

Reviewer: ```suggestion
    Allow a client to know which devices are polled
```

  * `DevPollStatus`:

>     Allow a client to precisely knows the polling status for a device.

Reviewer: ```suggestion
    Allow a client to precisely know the polling status for a device.
```

* The Polling configuration defined by a Client SHALL be stored in database and applied upon a Device initialisation.

* For each request from the Polling list, the Device Server SHALL store in a Cache at least the most recent response generated by the request.

* The depth of the Cache (the circular buffer size) SHALL be configurable.
  

> * For any Request, a Clinent MAY decide whether it want a response from the Cache or excute underlying request. The Device Server SHALL respond adequatly.

Reviewer: ```suggestion
* For any Request, a Client MAY decide whether it wants a response from the Cache or execute underlying request. The Device Server SHALL respond adequately.
```
